### PR TITLE
Nette 3.0

### PR DESCRIPTION
**Pull request by haltuf**
Rewrite to ConfigSchema + fix incompatibilities with Kdyby\Doctrine.
A necessary prerequisite for updating Kdyby\Doctrine to support Nette v3.0

### CODE REVIEW

File: src/DI/AnnotationsExtension.php [hunk 1]
```diff
@@ -54,8 +68,8 @@ public function loadConfiguration(): void
 			->setClass(Reader::class)
 			->setFactory(CachedReader::class, [
 				$this->prefix('@reflectionReader'),
-				Helpers::processCache($this, $config['cache'], 'annotations', $config['debug']),
-				$config['debug'],
+				Helpers::processCache($this, $config['cache'], 'annotations', $config['debug'] && $this->debugMode),
+				$config['debug'] && $this->debugMode,
```
Comment (enumag) [inline src/DI/AnnotationsExtension.php]: is this really how this is supposed to work in Nette 3? No way to enable debugMode for cache in production?

**Comment by enumag** [comment]
ping @haltuf 

Also check https://github.com/Kdyby/Annotations/pull/11 if there is anything there we should use.

File: src/DI/AnnotationsExtension.php (hunk 1 above)
Comment (haltuf) [reply src/DI/AnnotationsExtension.php]: You're right, stupid me - we actually do not need global debugMode at all. Changed.

**Comment by haltuf** [comment]
> Also check #11 if there is anything there we should use.

@enumag I don't think so, the PR is based on (IMO) wrong understanding of the changes in Nette/DI in 3.0 version. ::validateConfig is deprecated and renaming the method to the getModifiedConfig() is quite a non-standard approach.

**Comment by enumag** [comment]
Thank you!

**Pull request merged**